### PR TITLE
FW-589. Mercator project crashes when receiving wrong HDLC frame

### PR DESCRIPTION
**Pull request by keomabrun**

### CODE REVIEW

**Comment by twatteyne** [comment]
Is that what caused the misterious problems in IoT-lab?

**Comment by keomabrun** [comment]
I am not sure of that no. But the code should no crash if it occurs anyway.

**Comment by twatteyne** [comment]
ok

File: kernel/openos/scheduler.c [hunk 1]
```diff
@@ -52,9 +52,7 @@ void scheduler_start() {
          pThisTask->next          = NULL;
          scheduler_dbg.numTasksCur--;
       }
-      debugpins_task_clr();
```
Comment (changtengfei) [inline kernel/openos/scheduler.c]: The debugpins functions here shouldn't be deleted as it's used for other board.

File: kernel/openos/scheduler.c [hunk 2]
```diff
@@ -52,9 +52,7 @@ void scheduler_start() {
          pThisTask->next          = NULL;
          scheduler_dbg.numTasksCur--;
       }
-      debugpins_task_clr();
       board_sleep();
-      debugpins_task_set();                      // IAR should halt here if nothing to do
```
Comment (changtengfei) [inline kernel/openos/scheduler.c]: Same as previous one.

**Comment by changtengfei** [comment]
@keomabrun cool! I left two tiny comments. After they are solved, I will merge the code..

**Pull request merged**